### ISSUE TITLE
Inconsistent/crude death effects

### ISSUE THREAD

**Issue by LMH0013** [SuperTux/supertux#320]
While I am glad to see more interesting death effects, there have been aspects of the current implementation which have bothered me from the start.  I thought that maybe I was being overly critical, letting the physicist in me lament over the non-physical parts despite the fact that it is a game about a penguin fighting animate ice and snow.  However, I had to explain what was happening to my children for the animations confused them-- I guess they were able to intuitively identify something was wrong.  So I'd like to request some refinements.

First the melting animations are wickedly inconsistent:
- blue iceblocks melt to blue water, pink ice/snow melt to pink water, white snow melt to blue water
- enemies melt into an orb of water unless they happen to be wearing a hat, then they melt into a series of smaller blobs
- if the badguy is walking on flat ground meltwater is absorbed by ground, otherwise water falls below the ground
- melting badguys make the water sound when hit, although sometimes they make the hissing sound as well, burning badguys do not make any sound when hit
- all badguy accessories disappear upon melting excepting spiky hats

Then there are things that look confusing:
- shoot a badguy at the edge of a platform and watch the water flatten out into the air
- melting spiky/jumpy results in a dangerous-looking but harmless shell that looks only slightly different from the badguy itself being left behind, this then magically disappears

Finally there are the things that look bad but honestly might be more my OCD:
- badguys are simply replaced with globs of water, so we really do not have death animations so much as sudden transitions to another animated object
- kick an iceblock and then shoot it, it will instantly stop and melt in place, horizontal motion is not preserved during melting, same goes for flying showballs
- walking badguys don't melt so much as ooze into the ground, water is not that viscus
- there's way too much water upon melting snowballs, snow is typically considerably less dense than water (could even be a factor of 10 in some cases).  Currently we're making mass because the water ball is roughly the same size as the snowball which I find bothersome.  Don't believe me?  Load up a bucket with snow and let it melt, you will not have a filled bucket once it does.
- have you ever seen water drop as a perfect sphere that sticks all together?  Not even raindrops are actually spherical, much less a large glob of water.  Plus the surface tension is not enough to maintain a single shape.  In this sense, I actually prefer the animations for spiky/jumpy.  

I think that the inconsistent and confusing aspects really need to be addressed.  I could probably live with the rest, but I'd be grumbling all the way.  That's not to say I'm meaning to be overly-critical.  Turns out I tried to make my own death animations but they were so bad I never pushed them, so I understand the difficult subtleties involved especially when animating fluids.  For what it's worth, I think the burning deaths look acceptable, and I really enjoyed the splashes when an orb of meltwater hit the ground.  Hopefully this list can aid in improving the death animations even more.

**Comment by tobbi** [comment]
This should definitely be improved. I will assign Hume to it, since he was the one who created these kind of effects in the first place.

**Comment by Hume2** [comment]
The water could be clearer, but it will be barely visible then, because there is no light refraction so far. The accessories of badguys shouldn't melt, that's true. Burning badguys make sound too, but it's very quiet. The sound could be changed to something else, but I'm not the right one to do it. I've noticed too that the shell that jumpys and spikys leave look to dangerous, there are two options: make it dangerous or redraw it so it will be obvious that it isn't dangerous, for example it could rust in that powerful fire. For the melting animation, we can't apply the normal melting process, because this melting is very fast. To do it really precise would also require much more realistic water physics. I'm not sure about lags, but I can look arround on the web for a suitable water simulation algorythm.

**Comment by LMH0013** [comment]
Instead of clear, you could go semi-transparent.

Maybe we should set up a mechanic to easily produce arbitrary particles on death.  For example, have the sprite contain several "drop" actions, if they exist, then they get drawn when burned/melted.  The graphics drawn could then be things like shoes, bows, hats, flying platform, etc.  But keep it as general as possible so that each badguy does not need new code.

I take it that the "flame" sound is then used?  It is very quiet... What we probably need a sound person to generate a unique fire sound to denote burning.  We could at least get the code in place, change the name of the sound made and duplicate an existing sound file as a filler.

I wouldn't make the shells dangerous, I've experimented with that sort of thing and it's really annoying to play.  Rusting would be fine, although in all honesty you could even have it burn.  Yeah I know, burning metal makes no sense, but I'm not sure how to do rusting without causing player confusion at least fireball -&gt; fire follows on a very basic level.

The melting is the tricky part...  I've always just gone with sublimation, but admittedly that's the coward's way out, water is much more ambitious.  If you don't want to get into the specifics of a simulation, you might get away with generating water particles semi-randomly.  You'd sadly loose the big splash...  or go straight to the splash, that might work.  In fact it already happens when a badguy is walking on a slope-- kill two birds with one stone, make the inconsistency be the default.

---- On Wed, 06 Jan 2016 07:17:43 -1000 Hume2 &lt;notifications@github.com&gt; wrote ---- 

The water could be clearer, but it will be barely visible then, because there is no light refraction so far. The accessories of badguys shouldn't melt, that's true. Burning badguys make sound too, but it's very quiet. The sound could be changed to something else, but I'm not the right one to do it. I've noticed too that the shell that jumpys and spikys leave look to dangerous, there are two options: make it dangerous or redraw it so it will be obvious that it isn't dangerous, for example it could rust in that powerful fire. For the melting animation, we can't apply the normal melting process, because this melting is very fast. To do it really precise would also require much more realistic water physics. I'm not sure about lags, but I can look arround on the web for a suitable water simulation algorythm.

**Comment by Karkus476** [comment]
@LMH0013 To be quite honest, the burn animation looks dangerous...

**Comment by Hume2** [comment]
The result of burning iron is rust. Anyway I calculated that the fireball could be made of coal near its melting point, so it might be senseful.

**Comment by Hume2** [comment]
The result of burning iron is rust. Anyway I calculated that the fireball could be made of coal near its melting point, so it might be senseful.

**Comment by Karkus476** [comment]
@Hume2 There's no point trying to explain what goes on in SuperTux, it's a game.

**Comment by misterhat** [comment]
>What we probably need a sound person to generate a unique fire sound to denote burning. We could at least get the code in place, change the name of the sound made and duplicate an existing sound file as a filler.

http://opengameart.org/content/fire-evil-spell or http://opengameart.org/content/magic-sfx-sample (the first clip of each)?

**Comment by LMH0013** [comment]
@Karkus476 You're right about the burning animation, but at least it doesn't look like the living badguy that does hurt Tux.  The burning effect could be toned down.

@Hume2 While burning iron produces iron oxide, which is the same product as rusting, the processes are quite different.  What I wouldn't do is make the spike helm spontaneously rust, that's going to cause confusion.  A toned down burn leaving behind a rust colored, disheveled ashen helm that then crumbles seems like a viable possibility.

@misterhat The first part of the fire-evil-spell sounds perfect to me.  In addition, we could use the ice blast (second sound in the magic-sfx-sample) as the sound for freezing enemies which is also currently missing (might as well work on that at the same time).  Thanks for the suggestion!

What do others think about the sounds?

**Comment by Hume2** [comment]
It was decided that only some of those issues will be fixed, so I'll close this after I'll fix all of them.

**Comment by tobbi** [comment]
@Hume2: The sound above is CC-BY. Did you add the credits for that?

**Comment by Hume2** [comment]
@tobbi It's in the ogg metadata, nowhere else yet.

**Comment by tobbi** [comment]
@Hume2 Please add it to the credits and the AUTHORS file in data/

**Comment by Alasdairbugs** [comment]
Closed as stale, most enemy animations and such are being replaced so this needs to be re-submitted with clearer notes on whats inconsistent.
